### PR TITLE
Add z-index to main container

### PR DESCRIPTION
**Pull request by emplums**
Adds a `z-index: 0` to the main layout container, to fix some issues with the `position: sticky` header messing with stacking context. 

Fixes: #96

### CODE REVIEW

**Comment by colebemis** [comment]
@emplums This is super minor but should we be using be `css` prop instead of the inline `style` prop here?

Also, don’t we usually camelCase css properties? (i.e. `zIndex` vs `“z-index”`)

**Comment by emplums** [comment]
@colebemis I used the inline style just because I wasn't sure if we had the style components babel plugin setup or not! If we do I can switch it back :)

**Comment by colebemis** [comment]
In this case, I don't think the `style` vs `css` prop makes a huge difference. However, I think we should be consistent about how we apply one-off styles, and the `css` prop is a more flexible way to apply styles. @mxstbr wrote a good blog post about inline styles here: https://mxstbr.blog/2016/11/inline-styles-vs-css-in-js/

Also, the `styled-components` babel plugin is set up by the `styled-components` Gatsby plugin: https://github.com/primer/doctocat/blob/master/theme/gatsby-config.js#L13

File: theme/src/components/layout.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ function Layout({children, pageContext}) {
     <Flex flexDirection="column" minHeight="100vh">
       <Head title={title} description={description} />
       <Header />
-      <Flex flex="1 1 auto" flexDirection="row">
+      <Flex flex="1 1 auto" flexDirection="row" css={{'zIndex': '0'}}>
```
Comment (colebemis) [inline theme/src/components/layout.js]: [Minor] I don't think these need to be in quotes :)

```suggestion
      <Flex flex="1 1 auto" flexDirection="row" css={{zIndex: 0}}>
```

**Pull request merged**